skyportal/handlers/api/reminder.py

```diff
@@ -0,0 +1,900 @@
+                )
+            except AccessError:
+                return self.error(
+                    f'Could not access GcnEvent {gcn_event.id}.', status=403
+                )
+            for user in users:
+                reminder = ReminderOnGCN(
+                    text=reminder_text,
+                    gcn_id=gcn_event.id,
+                    author=author,
+                    groups=groups,
+                    bot=is_bot_reminder,
+                    next_reminder=next_reminder,
+                    reminder_delay=reminder_delay,
+                    number_of_reminders=number_of_reminders,
+                    users=users,
+                )
+        elif associated_resource_type.lower() == "shift":
+            shift_id = resource_id
+            try:
+                shift = Shift.get_if_accessible_by(
+                    shift_id, self.current_user, raise_if_none=True
+                )
+            except AccessError:
+                return self.error(f'Could not access Shift {shift.id}.', status=403)
+            for user in users:
+                reminder = ReminderOnShift(
+                    text=reminder_text,
+                    shift_id=shift.id,
+                    author=author,
+                    groups=groups,
+                    bot=is_bot_reminder,
+                    next_reminder=next_reminder,
+                    reminder_delay=reminder_delay,
+                    number_of_reminders=number_of_reminders,
+                    users=users,
+                )
+        else:
+            return self.error(f'Unknown resource type "{associated_resource_type}".')
+
+        if associated_resource_type.lower() == "sources":
+            text_to_send = f"*@{self.associated_user_object.username}* created a reminder on *{obj_id}*"
+            url_endpoint = f"/source/{obj_id}"
+        elif associated_resource_type.lower() == "spectra":
+            text_to_send = f"*@{self.associated_user_object.username}* created a reminder on *{spectrum_id}*"
+            url_endpoint = f"/source/{spectrum_id}"
+        elif associated_resource_type.lower() == "gcn_event":
+            text_to_send = f"*@{self.associated_user_object.username}* created a reminder on *{gcnevent_id}*"
+            url_endpoint = f"/gcn_events/{gcnevent_id}"
+        elif associated_resource_type.lower() == "shift":
+            text_to_send = f"*@{self.associated_user_object.username}* created a reminder on *shift {shift_id}*"
+            url_endpoint = "/shifts"
+        else:
+            return self.error(f'Unknown resource type "{associated_resource_type}".')
+
+        for user in users:
+            DBSession().add(
+                UserNotification(
+                    user=user,
+                    text=text_to_send,
+                    notification_type="mention",
+                    url=url_endpoint,
+                )
+            )
+
+        DBSession().add(reminder)
```

Comment:
Might be nice to have that logic in a separate function so we can create reminders from other handlers (like automatically creating a reminder when a user is added to a shift).